File: HelloWorldSDK/HelloWorld.m
//
//  HelloWorld.m
//  HelloWorldSDK
//
//  Created by peter.shih on 2019/7/23.
//  Copyright © 2019年 Peter.shih. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "HelloWorld.h"

@implementation HelloWorld

+(NSString*)helloWithName:(NSString *)name {
    return [NSString stringWithFormat:@"%@, %@", @"Hello", name];
}

-(NSString*)hiWithName:(NSString *)name {
    return [NSString stringWithFormat:@"%@, %@", @"Hi", name];
}

@end

File: HelloWorldSDK/HelloWorld.swift
//
//  HelloWorld.swift
//  HelloWorldSDK
//
//  Created by peter.shih on 2019/8/26.
//  Copyright © 2019年 Peteranny. All rights reserved.
//

public class HelloWorld {
    let greet = "Hello"

    public init() {}

    public func hello(to whom: String) -> String {
        return "\(greet) \(whom)"
    }
    
    public var helloColor: UIColor? {
        if #available(iOS 11.0, *) {
            return UIColor(named: "HelloColor", in: Bundle(for: HelloWorld.self), compatibleWith: nil)
        } else {
            return nil
        }
    }
}
